src/pyxir/runtime/backends/vai_rt/xir_api/dpu_func.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Copyright 2020 Xilinx Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <cstdlib>
+#include <cassert>
+#include <algorithm>
+#include <chrono>
+
+#include "pyxir/common/util.hpp"
+#include "dpu_func.hpp"
+
+namespace pyxir {
+namespace runtime {
+namespace vai_rt {
+
+DpuFunc::DpuFunc(XLayerHolder &xl, const std::string &build_dir) : KernelFunc(xl)
+{
+  std::vector<std::string> dpu_in_tensor_names = xl->bottoms;
+  std::vector<std::string> dpu_internal_in_tensor_names = xl->get_attr("input_names").get_strings();
+
+  in_tensor_names_ = dpu_internal_in_tensor_names;
+
+  std::vector<std::string> dpu_out_tensor_names = xl->get_attr("output_names").get_strings(); // xl->tops;
+  out_tensor_names_ = dpu_out_tensor_names;
+
+  std::unordered_map<std::string, std::string> rt_in_map =
+      xl->get_attr("rt_in_map").get_map_str_str();
+  std::unordered_map<std::string, std::string> rt_out_map =
+      xl->get_attr("rt_out_map").get_map_str_str();
+
+  pxDebug("Before DpuRunner init");
+  // Setup DPU runner
+  std::string model_path;
+  if (!build_dir.empty())
+  {
+    model_path = build_dir;
+  }
+  else
+  {
+    model_path = xl->get_attr("work_dir").get_string();
+  }
+  graph = xir::Graph::deserialize(model_path + "/xp0.xmodel");
```

Comment:
We should be able to make 'xp0' not hardcoded. I think xl->name should give you the name.